stdimage/models.py

```diff
@@ -31,31 +31,46 @@ class StdImageFieldFile(ImageFieldFile):
 
     def save(self, name, content, save=True):
         super(StdImageFieldFile, self).save(name, content, save)
-        render_variations = self.field.render_variations
-        if callable(render_variations):
-            render_variations = render_variations(
-                file_name=self.name,
-                variations=self.field.variations,
-                storage=self.storage,
-            )
-        if not isinstance(render_variations, bool):
-            msg = (
-                '"render_variations" callable expects a boolean return value,'
-                ' but got %s'
-                ) % type(render_variations)
-            raise TypeError(msg)
-        if render_variations:
-            self.render_variations()
+
+        # Skip rendering if render_variations set False
+        if isinstance(self.field.render_variations, bool) \
+           and not self.field.render_variations:
+            return
+
+        self.render_field_variations(self.name, self.field, False,
+                                     self.storage)
 
     @staticmethod
     def is_smaller(img, variation):
         return img.size[0] > variation['width'] \
             or img.size[1] > variation['height']
 
-    def render_variations(self, replace=False):
-        """Render all image variations and saves them to the storage."""
-        for _, variation in self.field.variations.items():
-            self.render_variation(self.name, variation, replace, self.storage)
+    @classmethod
+    def render_field_variations(cls, file_name, field, replace=False,
+                                storage=default_storage):
+        """
+        Force render field variations.
+
+        If field's render_variations is false then render with default
+        render_variation function.
+        """
+        if callable(field.render_variations):
```

Comment:
> Design wise render_variations is a boolean that decides whether or not to run the internal process. Passing a callable is just a nice side effect.
> They way you're currently splitting the two methods, does not reflect this idiom.

In my opinion it depends. Currently `render_field_variations` is the place where the decision is made but the meaning of the variable haven't change.
The reason I put it all there is I want get exactly the same result when I call `rendervariations` as when I recreate objects manually and I didn't want to break another behavior.
I think its right that these 3 sequences bellow should give the same result:
1. Create object. Modify `render_variations`. Call `rendervariations --replace`
2. Modify `render_variations`. Create object.
3. Create object. Modify `render_variations`. Delete object. Create object with same values.

I agree that this is not the most beautiful code but it fixes the bug and didn't change they way things already work. :)

As I said before I believe it's better to refactor `render_variations` into `render_variation` in future to make it all looks better!